enumerable.rb

```diff
@@ -0,0 +1,149 @@
+module Enumerable
+  def my_each
+    if block_given?
+      i = 0
+      while i < results.length
+        yield(results[i])
+        i += 1
+      end
+      results
+    else
+      results.to_enum
+    end
+  end
+
+  def my_each_with_index
+    if block_given?
+      i = 0
+      while i < results.length
+        yield(results[i], i)
+        i += 1
+      end
+      results
+    else
+      results.to_enum
+    end
+  end
+
+  def my_select
+    if block_given?
+      choice = []
+      results.my_each do |x|
+        if yield(x)
+          choice.push(x)
+        end
+      end
+      choice
+    else
+      results.to_enum
+    end
+  end
+
+  def my_all?(items = nil)
+    if block_given?
+      my_each { |x| return false unless yield x }
+    else
+      return my_all? { |obj| obj } unless items
+
+      if items.class == Regexp
+        my_each { |x| return false unless items.match?(x) }
+      elsif items.class == Class
+        my_each { |x| return false unless x.is_a? items }
+      else
+        my_each { |x| return false unless x == items }
+      end
+    end
+    true
+  end
+
+  def my_any?(items = nil)
+    result = false
+    if block_given?
+      my_each { |item| result = true if yield item }
+    elsif items
+      my_each { |item| result = true if itemstern?(item, items) }
+    else
+      my_each { |item| result = true if item }
+    end
+    result
+  end
+
+  def my_none?(items = nil)
+    result = true
+    if block_given?
+      my_each { |item| result = false if yield item }
+    elsif items
+      my_each { |item| result = false if itemstern?(item, items) }
+    else
+      my_each { |item| result = false if item }
+    end
+    result
+  end
+
+  def my_count(arg = nil)
+    count = 0
+    if block_given?
+      my_each { |x| count += 1 if yield(x) }
+    elsif arg
+      my_each { |x| count += 1 if x == arg }
+    else
+      count = length
+    end
+    count
+  end
+
+  def my_map
+    if block_given?
+      numbers = []
+      results.my_each do |x|
+        numbers << yield(x)
+      end
+      numbers
+    else
+      results.to_enum
+    end
+  end
+
+  def my_map_proc(&multiply_num)
+    if block_given?
+      numbers = []
+      results.my_each do |x|
+        numbers << multiply_num.call(x)
+      end
+      numbers
+    else
+      results.to_enum
+    end
+  end
+
+  def my_inject(*args)
+    result, sample = inj_param(*args)
+    arr = result ? to_a : to_a[1..-1]
+    result ||= to_a[0]
+    if block_given?
+      arr.my_each { |x| result = yield(result, x) }
+    elsif sample
+      arr.my_each { |x| result = result.public_send(sample, x) }
+    end
+    result
+  end
```

Comment:
- [ ] combines all elements of enum by applying a binary operation, specified by a block
- [ ] when a symbol is specified combines each element of the collection by applying the symbol as a named method

Reply:
can you give an example of one of the methods

Reply:
Can you give an example of what I need to do because I dont know what you want me to do

Reply:
The whole idea is for your methods to behave the same way as the original [enumerable methods](https://ruby-doc.org/core-2.7.0/Enumerable.html).
For example, 
```ruby
let(:range) { Range.new(5, 50) }
actual = range.my_inject(4) { |prod, n| prod * n }
expected = range.inject(4) { |prod, n| prod * n }
expect(actual).to eq expected
```
Should return true if your method `my_inject` works as expected.

Reply:
Ok